routes/main.js

```diff
@@ -14,6 +15,7 @@ module.exports = function (passport) {
 
 	router.get('/learn/vulnerability/:vuln', authHandler.isAuthenticated, function (req, res) {
 		var query_rating = req.query.securityRating ? req.query.securityRating : 0;
+		ratingState[req.params.vuln] = query_rating
```

Comment:
This is the point where the backend saves the currently checked rating to the ratingState module

Reply:
Use this instead:
var query_rating = req.query.securityRating ? req.query.securityRating : ratingState[req.params.vuln];